### PR TITLE
fix: request.getUri should respond include path and query values

### PR DESCRIPTION
**Pull request by sdelamo**
see: https://github.com/micronaut-projects/micronaut-aws/issues/1848

I added the same test to the core TCK I have added a test in the TCK https://github.com/micronaut-projects/micronaut-core/pull/9791

### CODE REVIEW

**Comment by sdelamo** [comment]
@timyates can you fix this. see approach: https://github.com/micronaut-projects/micronaut-aws/pull/1849

File: function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/ApiGatewayServletRequest.java [hunk 1]
```diff
@@ -108,6 +109,17 @@ protected ApiGatewayServletRequest(
 
     public abstract byte[] getBodyBytes() throws IOException;
 
+    protected static URI buildUri(
```
Comment (sdelamo) [inline function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/ApiGatewayServletRequest.java:112]: It misses javadoc

File: function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/payload2/APIGatewayV2HTTPEventServletRequest.java [hunk 2]
```diff
@@ -55,7 +55,11 @@ public APIGatewayV2HTTPEventServletRequest(
         super(
             conversionService,
             requestEvent,
-            URI.create(requestEvent.getRequestContext().getHttp().getPath()),
+            ApiGatewayServletRequest.buildUri(
+                requestEvent.getRequestContext().getHttp().getPath(),
```
Comment (driverpt) [inline function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/payload2/APIGatewayV2HTTPEventServletRequest.java:59]: Why not use `rawPath` and `rawQuery` ?

Comment (timyates) [reply function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/payload2/APIGatewayV2HTTPEventServletRequest.java:59]: What would be the advantage of using rawPath and rawQuery? 🤔

Comment (driverpt) [reply function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/payload2/APIGatewayV2HTTPEventServletRequest.java:58]: You don't need to parse and treat it

**Pull request merged**